### PR TITLE
Implement workflow endpoint

### PR DESCRIPTION
**Pull request by buchi**
- `GET /:path/workflow` (get workflow info)
- `POST /:path/workflow/:transition` (trigger workflow transition)

### CODE REVIEW

File: docs/source/_json/workflow_get.json [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+HTTP 200 OK
+content-type: application/json
+
+{
+  "history": [
+    {
+      "action": null, 
+      "actor": "test_user_1_", 
+      "comments": "", 
+      "review_state": "private",
```
Comment (lukasgraf) [inline docs/source/_json/workflow_get.json]: Can the current workflow state only be determined by looking at the most recent history entry? Or am I missing something?

File: docs/source/workflow.rst [hunk 2]
```diff
@@ -0,0 +1,19 @@
+.. include:: /alert-noindex.rst
```
Comment (lukasgraf) [inline docs/source/workflow.rst]: This shouldn't be included here (inserts the pre-alpha disclaimer and excludes page from google robots)

File: docs/source/_json/workflow_get.json (hunk 1 above)
Comment (buchi) [inline docs/source/_json/workflow_get.json]: The current workflow state will also be included in the object's GET response.
Regarding the workflow endpoint, i guess getting the complete history is enough. What else would you expect?

**Comment by lukasgraf** [comment]
Some minor points, LGTM otherwise 👍

File: docs/source/_json/workflow_get.json (hunk 1 above)
Comment (lukasgraf) [inline docs/source/_json/workflow_get.json]: I guess both having in `GET /obj` as well as `GET /obj/workflow` is somewhat redundant. It's just that the current workflow state is one of the infos I would expect from the `workflow` endpoint as a user.

**Pull request merged**